### PR TITLE
Initial commit for GCS connector 3.0.0 development

### PR DESCRIPTION
**Pull request by medb**

### CODE REVIEW

File: gcs/CHANGES.md [hunk 1]
```diff
@@ -1,4 +1,4 @@
-### 2.2.3 - 2021-XX-XX
+### 3.0.0 - 2022-XX-XX
```
Comment (mprashanthsagar) [inline gcs/CHANGES.md:1]: Do we want these changes to be in 3.0.0 only ? Should we add a separate section for 3.0.0 instead of replacing 2.2.3 ?

Comment (medb) [reply gcs/CHANGES.md:1]: No, they are going to be in 2.2.3 version released from the `branch-2.2.x` branch too: https://github.com/GoogleCloudDataproc/hadoop-connectors/blob/branch-2.2.x/gcs/CHANGES.md#223---2021-xx-xx

There is usually some duplication in release notes when we branch out development of the older version to its own branch.

Comment (mprashanthsagar) [reply gcs/CHANGES.md:1]: Ah ok, i didnt notice we already created a 2.2.x branch, thought we would lose the history

**Review by mprashanthsagar — APPROVED** [review]

File: bigquery/CHANGES.md [hunk 2]
```diff
@@ -1,4 +1,4 @@
-### 1.2.1 - 2021-XX-XX
+### 2.0.0 - 2022-XX-XX
 
 1.  Update all dependencies to latest versions.
```
Comment (jaceklaskowski) [inline bigquery/CHANGES.md:3]: nit: Add `the` to `latest`.

File: coverage/pom.xml [hunk 3]
```diff
@@ -7,11 +7,11 @@
   <parent>
     <groupId>com.google.cloud.bigdataoss</groupId>
     <artifactId>bigdataoss-parent</artifactId>
-    <version>2.2.3-SNAPSHOT</version>
+    <version>3.0.0-SNAPSHOT</version>
   </parent>
 
   <artifactId>coverage</artifactId>
-  <version>2.2.3-SNAPSHOT</version>
+  <version>3.0.0-SNAPSHOT</version>
```
Comment (jaceklaskowski) [inline coverage/pom.xml:14]: Are these versions related somehow? Can we introduce a maven property to keep them in sync?

File: bigquery/pom.xml [hunk 4]
```diff
@@ -22,7 +22,7 @@
   <parent>
     <groupId>com.google.cloud.bigdataoss</groupId>
     <artifactId>bigdataoss-parent</artifactId>
-    <version>2.2.3-SNAPSHOT</version>
+    <version>3.0.0-SNAPSHOT</version>
```
Comment (jaceklaskowski) [inline bigquery/pom.xml:25]: I'm curious if that's required with `relativePath`?

File: pom.xml [hunk 5]
```diff
@@ -80,8 +80,8 @@
 
     <gpg.skip>true</gpg.skip>
 
-    <bigdataoss.version>2.2.3-SNAPSHOT</bigdataoss.version>
-    <bigquery-connector.version>1.2.1-SNAPSHOT</bigquery-connector.version>
+    <bigdataoss.version>3.0.0-SNAPSHOT</bigdataoss.version>
+    <bigquery-connector.version>2.0.0-SNAPSHOT</bigquery-connector.version>
```
Comment (jaceklaskowski) [inline pom.xml:84]: Why are these properties not used in the other poms?

File: bigquery/pom.xml (hunk 4 above)
Comment (medb) [reply bigquery/pom.xml:25]: I think each component should have a version explicitly declared in Maven POM

File: pom.xml (hunk 5 above)
Comment (medb) [reply pom.xml:84]: Probably some legacy, we will clean up it in follow up.

File: coverage/pom.xml (hunk 3 above)
Comment (medb) [reply coverage/pom.xml:14]: Maven does not like properties in artifact definition version field.

File: bigquery/CHANGES.md (hunk 2 above)
Comment (medb) [reply bigquery/CHANGES.md]: Done.
